src/actions/apply_trainer_action.rs

```diff
@@ -792,3 +796,61 @@ fn clemont_effect(acting_player: usize, state: &State) -> (Probabilities, Mutati
         name == "Magneton" || name == "Heliolisk" || name == "Clemont's Backpack"
     })
 }
+
+fn serena_effect(acting_player: usize, state: &State) -> (Probabilities, Mutations) {
+    // Put a random Mega Evolution Pokémon ex from your deck into your hand.
+    // All Mega evolutions are ex by definition
+    pokemon_search_outcomes_with_filter_multiple(acting_player, state, 1, |card| card.is_mega())
+}
+
+fn quick_grow_extract_effect(acting_player: usize, state: &State) -> (Probabilities, Mutations) {
+    // Choose 1 of your [G] Pokémon in play. Put a random [G] Pokémon from your deck
+    // that evolves from that Pokémon onto that Pokémon to evolve it.
+    // Similar to rare candy but automatic random evolution from deck
+
+    // Find all Grass Pokemon in play that have valid evolutions in deck
+    let mut evolution_choices: Vec<(usize, Card)> = vec![];
+
+    for (in_play_idx, pokemon) in state.enumerate_in_play_pokemon(acting_player) {
+        if pokemon.get_energy_type() != Some(EnergyType::Grass) || pokemon.played_this_turn {
+            continue;
+        }
+
+        // Find Grass evolutions in deck that evolve from this Pokemon
+        let pokemon_name = pokemon.get_name();
+        for deck_card in state.decks[acting_player].cards.iter() {
+            if let Card::Pokemon(deck_pokemon) = deck_card {
+                if deck_pokemon.energy_type == EnergyType::Grass
+                    && deck_pokemon.evolves_from.as_ref() == Some(&pokemon_name)
```

Comment:
Can we use the `can_evolve_into` function in the `hook.rs` package? Not relevant now in theory since the special cases are just for Eevee's, but into the future it might change so that can help make this future proof.